proxy_handler_302_test.py

```diff
@@ -1,27 +1,36 @@
 #!/usr/bin/env python
 
-# NOTE: subprocess.run requires Python 3.5+
-import subprocess
-from subprocess import run
+from subprocess import Popen
+from subprocess import PIPE
+import sys
+
+# We start off with the assumption that the test will succeed
+ec = 0
 
 # Test proxy server 
 print('DEBUG: Testing 302 response for proxy server!')
-proxy_server_process = subprocess.Popen(['./lightning', 'proxy_config'])
+proxy_server_process = Popen(['./lightning', 'proxy_config'])
 
 # Store contents of request to www.ucla.edu 
-p = subprocess.Popen(["curl", "www.ucla.edu"], stdout=subprocess.PIPE, stderr=subprocess.PIPE)
+p = Popen(["curl", "www.ucla.edu"], stdout=PIPE, stderr=PIPE)
 expected_proxy_response = p.stdout.read()
-actual_proxy_response = run(['http', 'localhost:3030/reverse_proxy_redirect'], stdout=subprocess.PIPE)
-
-if (actual_proxy_response.returncode != 0):
-    print('FAILED: proxy server encountered an error')
+actual_proxy_response = Popen(['http', 'localhost:3030/reverse_proxy_redirect'], stdout=PIPE)
 
-if (actual_proxy_response.stdout != expected_proxy_response):
+if (actual_proxy_response.communicate()[0].decode('utf-8') != expected_proxy_response.decode('utf-8')):
     print('FAILED: proxy server received a non-matching proxy response')
-    print('Expected proxy response: \n%s' % expected_proxy_response)
-    print('Completed proxy response: \n%s' % actual_proxy_response.stdout.decode('UTF-8'))
+    # print('Expected proxy response: \n%s' % expected_proxy_response)
```

Comment:
We should uncomment them. Done as part of 0dc29a77f2625be86ef6fa8340b8edef9fe39e74.